drivers/spi/spi_b_renesas_ra8.c

```diff
@@ -0,0 +1,711 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra8_spi_b
+
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+#include <instances/r_dtc.h>
+#include <instances/r_spi_b.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ra8_spi_b);
+
+#include "spi_context.h"
+
+#if defined(CONFIG_SPI_B_INTERRUPT)
+void spi_b_rxi_isr(void);
+void spi_b_txi_isr(void);
+void spi_b_tei_isr(void);
+void spi_b_eri_isr(void);
+#endif
+
+struct ra_spi_config {
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct ra_spi_data {
+	struct spi_context ctx;
+	uint8_t dfs;
+	struct st_spi_b_instance_ctrl spi;
+	struct st_spi_cfg fsp_config;
+	struct st_spi_b_extended_cfg fsp_config_extend;
+#if CONFIG_SPI_B_INTERRUPT
+	uint32_t data_len;
+#endif
+#if defined(CONFIG_SPI_B_RA_DTC)
```

Comment:
Is it possible for someone to use DTC only on RX or TX?
If yes, we should use tx-dtc and rx-dtc DT properties.
If no, should we consider use only one property for DTC in DT (e.g. use_dtc)?

Reply:
@ydamigos , User are able to to use only tx dtc or rx dtc.
`#ifndef CONFIG_SPI_B_RA_DTC
#define RA_SPI_B_DTC_STRUCT_INIT(index)
#define RA_SPI_B_DTC_INIT(index)
#else
#define RA_SPI_B_DTC_INIT(index)                                                                   \
	do {                                                                                       \
		if (DT_INST_PROP_OR(index, rx_dtc, false)) {                                       \
			ra_spi_data_##index.fsp_config.p_transfer_rx =                             \
				&ra_spi_data_##index.rx_transfer;                                  \
		}                                                                                  \
		if (DT_INST_PROP_OR(index, tx_dtc, false)) {                                       \
			ra_spi_data_##index.fsp_config.p_transfer_tx =                             \
				&ra_spi_data_##index.tx_transfer;                                  \
		}                                                                                  \
	} while (0)`